src/main/scala/cromwell/engine/package.scala

```diff
@@ -26,6 +32,40 @@ package object engine {
     def randomId() = WorkflowId(UUID.randomUUID())
   }
 
+  /**
+   * Constructs a representation of a particular workflow invocation.  As with other
+   * case classes and apply() methods, this will throw an exception if it cannot be
+   * created
+   */
+  case class WorkflowDescriptor(id: WorkflowId, sourceFiles: WorkflowSourceFiles) {
+    val workflowOptions = Try(sourceFiles.workflowOptionsJson.parseJson) match {
+      case Success(options: JsObject) => WorkflowOptions.fromJsonObject(options).get // .get here to purposefully throw the exception
+      case Failure(ex) => throw ex
+    }
```

Comment:
Maybe a more idiomatic forcomp?  Also this doesn't check `parseJson` returning something that isn't a `JsObject`.